src/main/scala/viper/carbon/modules/impls/DefaultFuncPredModule.scala

```diff
@@ -302,10 +318,13 @@ with DefinednessComponent with ExhaleComponent with InhaleComponent {
    */
   private def transformFuncAppsToLimitedOrTriggerForm(exp: Exp, heightToSkip : Int = -1, triggerForm: Boolean = false): Exp = {
     def transformer: PartialFunction[Exp, Option[Exp]] = {
-      case FuncApp(recf, recargs, t) if recf.namespace == fpNamespace && (heightToSkip == -1 || heights(recf.name) <= heightToSkip) =>
+      case FuncApp(recf, recargs, t) if recf.namespace == fpNamespace &&
+        (heightToSkip == -1 || heights(if (recf.name.endsWith(limitedPostfix)) recf.name.dropRight(limitedPostfix.length) else recf.name) <= heightToSkip) =>
```

Comment:
It would be nice to have a comment here saying why we need to potentially remove the limited postfix here before looking up the name in `heights`.

Also, are we assuming here that `limitedPostfix` is a suffix that cannot appear in a Viper function? If so, it might make sense to add a comment here saying that we are making this assumption and adding a comment to the `limitedPostfix` field ([link](https://github.com/viperproject/carbon/blob/b2230000f7364ea23cbb1ce16cb16fc130e3f6fe/src/main/scala/viper/carbon/modules/impls/DefaultFuncPredModule.scala#L58)) stating that it should be a suffix that cannot appear in Viper functions. I think ' is indeed not allowed in Viper prgorams